src/source/PeerConnection/Rtcp.c

```diff
@@ -73,6 +102,15 @@ STATUS onRtcpPacket(PKvsPeerConnection pKvsPeerConnection, PBYTE pBuff, UINT32 b
                         rttPropDelayMsec = KVS_CONVERT_TIMESCALE(rttPropDelay, DLSR_TIMESCALE, 1000);
                         DLOGD("RTCP_PACKET_TYPE_RECEIVER_REPORT rttPropDelay %u msec", rttPropDelayMsec);
                     }
+
+                    MUTEX_LOCK(pTransceiver->sender.statsLock);
+                    pTransceiver->sender.remoteInboundRtpStreamStats.reportsReceived++;
+                    pTransceiver->sender.remoteInboundRtpStreamStats.fractionLost = (DOUBLE) fractionLost / 255.0;
+                    pTransceiver->sender.remoteInboundRtpStreamStats.roundTripTimeMeasurements++;
+                    pTransceiver->sender.remoteInboundRtpStreamStats.totalRoundTripTime += rttPropDelayMsec;
+                    pTransceiver->sender.remoteInboundRtpStreamStats.roundTripTime = rttPropDelayMsec;
+                    MUTEX_UNLOCK(pTransceiver->sender.statsLock);
```

Comment:
@MushMal 
locking code turned out to be not bad at all

Reply:
Yeah, indeed, it's not as bad as one would expect - your approach was fantastic with evaluating local vars which later get set in one go.

I think the only thing that's missing is freeing the mutex

Reply:
Just approved the changes. THANK YOU!!!